scanner/matcher/updater/vuln/updater.go

```diff
@@ -284,6 +309,11 @@ func (u *Updater) runUpdate(ctx context.Context) error {
 	if err := u.metadataStore.SetLastVulnerabilityUpdate(ctx, timestamp); err != nil {
 		return err
 	}
+
+	if u.initialized.CompareAndSwap(false, true) {
+		zlog.Info(ctx).Msg("finished initial updater run: setting updater to initialized")
```

Comment:
I think there may be a race condition where scanner calls`SetLastVulnerabilityUpdate`, then a request to get vulns comes in, the service calls `Initialized` which calls `GetLastVulnerabilityUpdate` and then sets `initialized` to `true`, and then we try this `CompareAndSwap`.

That all being said, this just affects the logs which we output. In this scenario, we'll see 

`previous run exists: setting updater to initialized` instead of `finished initial updater run: setting updater to initialized`. Is this ok with you/expected?

Maybe it'd be best to just log the same thing in both functions?

Reply:
As you noted, that is OK. Having two different log messages will not be confusing, IMO. It actually makes it explicit what happened.